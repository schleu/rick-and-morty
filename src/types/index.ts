  
  export const status = ["Alive", "Dead", "unknown"] as const
  export type Status = typeof status[number];
  
  export const gender = ["Female", "Male", "Genderless",  "unknown"] as const
  export type Gender = typeof gender[number]
  
  export interface iCharacter  {
    "id": number,
    "name": string,
    "status": Status,
    "species": string,
    "type": string,
    "gender": Gender,
    "origin": {
        "name": string,
        "url": string,
    },
    "location": {
        "name": string,
        "url": string,
    },
    "image": string;
    "episode": string[],
    "url": string,
    "created": string,
  }

  export interface iFilterCharacter {
    name?:string;
    status?:Status;
    gender?:Gender;
    species?:string;
    type?:string;
  }

  interface iFilterEpisode{
    name:string;
    code: number;
  }


export  interface iFilter{
    page?:number;
    character?:iFilterCharacter;
    epsode?:iFilterEpisode
  }

export interface response {
    info:{
      "count": number,
      "pages": number,
      "next": string,
      "prev": string
  },
    results:iCharacter[]
  }



  export interface iEpisode{
    id:number;
    name:string;
    air_date:string;
    episode: string;
    characters: string[];
    url: string;
    created:string;
  }