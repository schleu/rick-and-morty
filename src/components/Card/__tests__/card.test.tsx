import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'

import { iCharacter } from '../../../types/index'
import { Card } from '../index'


const character:iCharacter = {
  "id": 1,
  "name": "Rick Sanchez",
  "status": "Alive",
  "species": "Human",
  "type": "",
  "gender": "Male",
  "origin": {
      "name": "Earth (C-137)",
      "url": "https://rickandmortyapi.com/api/location/1"
  },
  "location": {
      "name": "Citadel of Ricks",
      "url": "https://rickandmortyapi.com/api/location/3"
  },
  "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
  "episode": [
      "https://rickandmortyapi.com/api/episode/1",
      "https://rickandmortyapi.com/api/episode/2",
      "https://rickandmortyapi.com/api/episode/3",
  ],
  "url": "https://rickandmortyapi.com/api/character/1",
  "created": "2017-11-04T18:48:46.250Z"
}

describe('<Card />', () => {
  beforeEach(()=>{
    render(<Card character={character} />);
  })

  it('snapshot', () => {
    const { container } = render(<Card character={character}/>)
    expect(container).toMatchSnapshot()
  })

  it('renders a character name', () => {
    
    const characterName = screen.getByRole('heading', {
      name: "Rick Sanchez",
    })

    expect(characterName).toBeInTheDocument()
  })
  
  it('renders a character species', () => {
    const text = screen.getByText(character.species)
    expect(text).toBeInTheDocument()
  })

  it('renders a character gender', () => {
    const text = screen.getByText(character.gender)
    expect(text).toBeInTheDocument()
  })

  it('renders a character status', () => {
    const text = screen.getByText(character.status)
    expect(text).toBeInTheDocument()
  })

})