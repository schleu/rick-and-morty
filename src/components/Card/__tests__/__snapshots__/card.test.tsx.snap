// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> snapshot 1`] = `
<div>
  <div
    class="card"
    data-testid="card"
  >
    <a
      href="/character/1"
    >
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="150"
        src="/_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F1.jpeg&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F1.jpeg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F1.jpeg&w=640&q=75 2x"
        style="color: transparent;"
        width="200"
      />
    </a>
    <div
      class="cardDescription"
    >
      <div
        class="cardTitle"
      >
        <h3>
          Rick Sanchez
        </h3>
        <button
          class="favorite"
          type="button"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height="20"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style="color: transparent;"
            width="20"
          />
        </button>
      </div>
      <p>
        <span>
          Human
        </span>
         - 
        <span>
          Male
        </span>
      </p>
      <p
        class="characterStatus"
      >
        <span
          class="characterStatusPointer"
        />
        <span>
          Alive
        </span>
      </p>
    </div>
  </div>
</div>
`;
